### DISCUSSION TITLE
Relationship between collections

### DISCUSSION THREAD

**Discussion by cloudcompute** [directus/directus#12930] [Feature Requests]
I have difficulty in specifying/understanding the different types of relationships that exist between two relations while trying to build them using the Directus UI.

For example, a Blog collection can have one or more Author (another collection). This means that the Blog's primary key should exist as a column in the Author collection (as a Foreign key).

While trying to specify it in the UI, I cannot do it. Inside the Author collection, create a field of the type One to Many, select the  "Related Collection" to Blog, in the corresponding Foreign Key field, the "id" field is greyed out (disabled) while other ordinary columns like blog_title, one can select. I am wondering how a blog title could be a foreign key, why does it not let me select the "id" as a primary key, am I missing something?

Couldn't we keep the UI simple.. something like this:

Let us try to visualise it by creating an altogether new UI window which is solely used to specify the relationship between different collections:

a. List all collections
b. User selects two collections. A ```drawer``` gets opened where four elements are displayed:
  - One-One
  - One-Many
  - Many-One
  - Many-Many

All four have two radio buttons Or a single drop-down with the values:
- Unidirectional
- Bidirectional 

Alternatively, a simple checkbox "Make the relation bidirectional" meaning the system assumes that a relation is unidirectional by default. Checking it will make the relation bidirectional.

For example, this is shown in the UI for a one-to-many relation:
Blog ----> Author

The user checks the box (or change the radio button or a value in the drop down). And the UI displays:
Blog <----> Author

There is no need to tell the user about what is happening behind the scenes i.e. no need to display the text like "Foreign key" etc. in the UI. He might not be from a technical background.

**Comment by rijkvanzanten** [comment]
> For example, a Blog collection can have one or more Author (another collection). This means that the Blog's primary key should exist as a column in the Author collection (as a Foreign key).

Not necessarily. In this particular case, you're looking for a Many-to-Many (M2M), as every blog can have multiple authors, and every author can have multiple blog posts. When configuring a many to many, you don't have to select any related foreign keys, as everything can be auto-generated 👍🏻 

> While trying to specify it in the UI, I cannot do it. Inside the Author collection, create a field of the type One to Many, select the "Related Collection" to Blog, in the corresponding Foreign Key field, the "id" field is greyed out (disabled) while other ordinary columns like blog_title, one can select. I am wondering how a blog title could be a foreign key, why does it not let me select the "id" as a primary key, am I missing something?

When creating a One-to-Many, you're asked to provide what related column to store the current item's primary key in. I think you're mistaken between the Many-to-One (M2O) and One-to-Many (O2M) in this particular setup, as it sounds like you were trying to configure a M2O field from Author to Blog.

> There is no need to tell the user about what is happening behind the scenes i.e. no need to display the text like "Foreign key" etc. in the UI. He might not be from a technical background.

I disagree with this. There's definitely room for abstracting away a lot of more technical concepts in the happy path, but I wouldn't want to prevent technical users from configuring things exactly according to their needs and requirements 👍🏻

**Comment by cloudcompute** [comment]
> In this particular case, you're looking for a Many-to-Many (M2M), as every blog can have multiple authors, and every author can have multiple blog posts. 

I just gave an example for One to Many. The point I wanted to say was why the system does not let me select "id" as a foreign key. Why does it grey-out while other fields which just cannot be a candidate for FK, the system says, you can select them but I wont let you selct "id"

**Reply by rijkvanzanten** [reply]
Cause ID is the primary key of the related collection, not the foreign key field where you'd want to store the related key in 🙂

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Comment by JawadSarfraz** [comment]
Can we create realation of Directus built-in User with on of the Collection?
